src/parser/core/modules/Medicated.ts

```diff
@@ -38,27 +38,36 @@ export class Medicated extends Analyser {
 		this.pot = {start: event.timestamp}
 	}
 
+	private stopAndSave(endTime: number) {
+		if (this.pot != null) {
+			this.pots.push({...this.pot, end: endTime})
+		}
+		this.pot = undefined
+	}
+
 	private offMedication(event: Events['statusRemove']) {
 		if (this.pot == null) {
 			throw new Error('potion instance not found')
 		}
-
-		this.pots.push({end: event.timestamp, ...this.pot})
-
-		this.pot = undefined
+		this.stopAndSave(event.timestamp)
 	}
 
 	private onComplete() {
-		const status = this.data.statuses.MEDICATED
+
+		// Add a pot that is still up before end of pull
+		if (this.pot != null && this.pot.end == null) {
+			this.stopAndSave(this.parser.pull.timestamp + this.parser.pull.duration)
+		}
 
 		if (this.pots.length > 0) {
 			const row = new SimpleRow({
 				label: 'Medicated',
 				order: -99,
 			})
 
+			const status = this.data.statuses.MEDICATED
```

Comment:
Thinking about it, this is easier to read if it's at the start, or otherwise inject it straight up where it's used.